sly.el

```diff
@@ -5250,6 +5250,32 @@ Full list of frame-specific commands:
 
 
 ;;;;; SLY-DB buffer creation & update
+
+(defcustom sly-db-focus-debugger 'always
+  "Control how debugger windows are displayed.
+ALWAYS: the debugger window is always focused.
+NEVER: the debugger window is never focused.
+REPL: only at the REPL."
+  :group 'sly-debugger
+  :type '(choice (const always)
+                 (const never)
+                 (const repl)))
+
+(defun sly-db-should-focus-debugger-p ()
+  "Decide whether to select the debugger window.
+Behavior depends on the current value of
+`sly-db-focus-debugger'."
+  (cl-ecase sly-db-focus-debugger
+    (always t)
+    (never nil)
+    (repl
+     (when-let (win (selected-window))
```

Comment:
Why do you need `when-let` here? Thw only case I can think of it returning nil (but I'm not sure) is in a `noninteractive` session, in which case you should check `noninteractive` instead. Same for `window-buffer`.

Reply:
BTW If you do need it, use `sly--when-let` instead. That way we won't break compatibility to older emacsen (just yet).

Reply:
I was thinking of a daemonized Emacs with no visible frames, but it seems selected-window always returns a window so the check is unnecessary.